examples/graphql-helix-auth0/index.ts

```diff
@@ -0,0 +1,127 @@
+/* eslint-disable no-console */
+import fastify from 'fastify';
+import { getGraphQLParameters, processRequest, renderGraphiQL, shouldRenderGraphiQL } from 'graphql-helix';
+import { envelop, useSchema } from '@envelop/core';
+import { useAuth0 } from '@envelop/auth0';
+import { makeExecutableSchema } from '@graphql-tools/schema';
+
+const schema = makeExecutableSchema({
+  typeDefs: /* GraphQL */ `
+    """
+    Describes the authentication object as proivided by Auth0.
+    """
+    type AuthenticationInfo {
+      """
+      String that uniquely identifies an authenticated user.
+      """
+      sub: String!
+    }
+
+    type Query {
+      """
+      The authentication information of the request.
+      """
+      authInfo: AuthenticationInfo
+    }
+  `,
+  resolvers: {
+    Query: {
+      authInfo(_source, _args, context) {
+        return context.auth0;
+      },
+    },
+  },
+});
+
+const auth0Config = {
+  domain: '{account_name}.{region}.auth0.com',
+  audience: 'http://localhost:3000/graphql',
+  clientId: '<insert this>',
+};
+
+const getEnveloped = envelop({
+  plugins: [
+    useSchema(schema),
+    useAuth0({
+      domain: auth0Config.domain,
+      audience: auth0Config.audience,
+      extendContextField: 'auth0',
+    }),
```

Comment:
Maybe we can add another example for adding `useGenericAuth` as well? (it can be combined with the `context.auth0` to resolve/validate the user) and also show how you can add auth using `@auth` directive?

Reply:
but it's a separate example, not sure if it's too advanced

Reply:
I think that should be a separate example.